### DISCUSSION TITLE
input-remapper install error

### DISCUSSION THREAD

**Discussion by jbealle** [sezanzeb/input-remapper#470] [Q&A]
Hello everyone I am new here.

I tried to install using the .deb file and got this error:
Error while installing package: dpkg-deb --control subprocess returned error exit status 2

So then I tried the manual steps under Ubuntu/Debian. In the last step I got the error below. Can anyone tell me what's happening?

- John

Get:1 /home/beallej/input-remapper/dist/input-remapper-1.5.0.deb input-remapper all 1.5.0 [158 kB]
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package input-remapper.
(Reading database ... 73462 files and directories currently installed.)
Preparing to unpack .../dist/input-remapper-1.5.0.deb ...
Unpacking input-remapper (1.5.0) ...
Setting up input-remapper (1.5.0) ...
Failed to disable unit: Unit file key-mapper.service does not exist.
Failed to stop key-mapper.service: Unit key-mapper.service not loaded.
Created symlink /etc/systemd/system/input-remapper.service → /usr/lib/systemd/system/input-remapper.service.
Created symlink /etc/systemd/system/default.target.wants/input-remapper.service → /usr/lib/systemd/system/input-remapper.service.
Processing triggers for mailcap (3.69) ...
Processing triggers for desktop-file-utils (0.26-1) ...
Processing triggers for dbus (1.12.20-2) ...

**Comment by jonasBoss** [comment]
Your log suggests everything installed fine. Does input-remapper work?

**Reply by jbealle** [reply]
Thank you so much for looking at this. But no, something definitely went wrong. These lines from the install don't seem right:

>debconf: delaying package configuration, since apt-utils is not installed
...
>Failed to disable unit: Unit file key-mapper.service does not exist.
>Failed to stop key-mapper.service: Unit key-mapper.service not loaded.

Here are the actions that don't work:

- the program doesn't appear in the Search/Launcher/Applications menu
- terminal command "sudo systemctl start input-remapper" doesn't do anything
- terminal command "sudo input-remapper-control --list-devices" doesn't do anything
- haven't seen any GUI at all

I am able to start the service with "sudo input-remapper-service" and configure the symbol names with "input-remapper-control --symbol-names". 

But then if I run the program with "input-remapper-gtk" I get this:

>Error executing command as another user: Not authorized
>This incident has been reported.
>ERROR: Failed to pkexec the helper, code 32512

Any ideas?

Thanks,
John

**Reply by jonasBoss** [reply]
> Failed to disable unit: Unit file key-mapper.service does not exist.
Failed to stop key-mapper.service: Unit key-mapper.service not loaded.

This is fine if you never had Input-Remapper installed before.
> terminal command "sudo systemctl start input-remapper" doesn't do anything

The command appears to do nothing but it should start inputremapper in the background.
If you run `ps -ax | grep input-remapper` you should see two processes:
```
$ ps -ax | grep input-remapper
   4143 ?        Ssl    0:00 /usr/bin/python3 /usr/bin/input-remapper-service
   4147 ?        S      0:00 /usr/bin/python3 /usr/bin/input-remapper-service
   4208 pts/1    S+     0:00 grep input-remapper
```

> But then if I run the program with "input-remapper-gtk" I get this:
>> Error executing command as another user: Not authorized
>> This incident has been reported.
>> ERROR: Failed to pkexec the helper, code 32512

You should be prompted for your password because the gui needs privileges in order to read the raw inputs form your devices.

If you just run `pkexec` in the terminal do you get a password prompt?
Can you run `input-remapper-gtk -d` and post the log here?
